crates/fj-host/src/lib.rs

```diff
@@ -260,22 +265,31 @@ impl Watcher {
     ///
     /// Returns `None`, if the model has not changed since the last time this
     /// method was called.
-    pub fn receive(&self) -> Option<fj::Shape> {
+    pub fn receive(&self, mut status: &mut StatusReport) -> Option<fj::Shape> {
+        status.update_status("Receiving updated shape...");
```

Comment:
Hey @devanlooches, I think this line is definitely one of the problems. The `receive` method is called in `run`, at the beginning of every iteration of the event loop. Meaning whatever the status used to be, at the beginning of the event loop (and therefore before every render) it is overwritten by this.

This is not the only problem. Other status updates seem to have a similar effect, and I haven't had time to understand everything completely. But hopefully this is enough to get you unstuck, so you can make more progress.

If you hit any more problems, always feel free to ask!

Reply:
Hi! Thanks for the help I'll see if I can figure out what I can do.

Reply:
So I have the full error showing up finally! But since it is the full compiler error it is very badly formatted:
<img width="166" alt="image" src="https://user-images.githubusercontent.com/50296057/182864644-51af26b7-e633-40c7-b715-ec64820422fb.png">
Do you have any thoughts on how to filter it?

Reply:
Sounds good. I think there is a scroll area container that we can use to wrap around the label [see here](https://docs.rs/egui/0.18.1/egui/containers/struct.ScrollArea.html).

Reply:
I will also update the docs for the StatusReport struct.

Reply:
Something like this works:

```rust
egui::ScrollArea::vertical().max_height(16.0)
    .show(ui, |ui| ui.label("Test label\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\ntest2"));
```

Reply:
Looks good! Looking forward to seeing that.